### ISSUE TITLE
types: add byte-comparable representation for all types

### ISSUE THREAD

**Issue by denesb** [scylladb/scylladb#19407]

**Comment by nyh** [comment]
But... The existing non-byte-ordered ordering (a conspicuous example is timeuuid ordering) is what we use for sorting sstables (**sorted** string table) and it's visible to users when scanning a partition in clustering key order. How can you change this order? Will your trie be scannable in the original non-byte order? I don't see how.

**Comment by michoecho** [comment]
> This kind of byte shuffling is indeed possible for timeuuid, I don't know about all other types (floating point, decimal, etc)

@nyh You might have missed the fact that cassandra implemented this (for all types) a while time ago.

**Comment by mykaul** [comment]
@denesb - what's missing (other than counters?) to close this issue?

**Comment by michoecho** [comment]
> [@denesb](https://github.com/denesb) - what's missing (other than counters?) to close this issue?

Collections.

**Comment by denesb** [comment]
There will be another PR adding byte-comparable representation to all frozen (only these are allowed in keys) collections. @lkshminarayanan is already working on it.

**Comment by denesb** [comment]
@lkshminarayanan is there anything left to do here?

**Comment by lkshminarayanan** [comment]
> [@lkshminarayanan](https://github.com/lkshminarayanan) is there anything left to do here?

Yes. I have to add [compatibility tests for date_type](https://github.com/scylladb/scylladb/pull/26039#discussion_r2353227030) and [remove the abstract type](https://github.com/scylladb/scylladb/pull/26039#discussion_r2353226765). We can close it once fixing these.

**Comment by lkshminarayanan** [comment]
Implemented across the following PRs/Commits :

- https://github.com/scylladb/scylladb/pull/23541
- https://github.com/scylladb/scylladb/pull/25603
- https://github.com/scylladb/scylladb/commit/bf90018b8ed171a2a2e3f16bbf18f45638ab8400
- https://github.com/scylladb/scylladb/pull/26206
- https://github.com/scylladb/scylladb/pull/26208
